xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/internal/template/InternalTemplateManager.java

```diff
@@ -894,4 +895,20 @@ public Template getTemplate(String templateName)
 
         return template;
     }
+
+    /**
+     * Create a DefaultTemplate with the given content and the given author.
+     *
+     * @param content the template content
+     * @param author the template author
+     * @return the template
+     */
+    public Template createStringTemplate(String content, DocumentReference author)
+    {
+        DefaultTemplate newTemplate = new DefaultTemplate(null);
```

Comment:
It would be cleaner to provide a dedicated StringTemplate taking content and author and generating a proper StringResource (probably extending AbstractInputSourceResource with a StringInputSource containing the passed content).

I don't even understand how your code can work since newTemplate.content should be null.